src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModelUnitTest.java

```diff
@@ -106,38 +277,66 @@ public void setupModel() {
-            tests.add(new Object[]{read, cigar1, ref, 3, Arrays.asList(0,0,0,0,0,0,0,0,0,0,0,0,0)});
-            tests.add(new Object[]{read, cigar2, ref, 3, Arrays.asList(1,1,1,1,1,1,0,0,0,0,0,0,0)});
+            tests.add(new Object[]{read, cigar1, null, ref, 3, 0, Arrays.asList(0,0,0,0,0,0,0,0,0,0,0,0,0)});
+            tests.add(new Object[]{read, cigar2, null, ref, 3, 0, Arrays.asList(1,1,1,1,1,1,0,1,1,1,1,0,0)}); //TODO this test was broken before without adjusting the offsets at all...
         }
         return tests.toArray(new Object[][]{});
     }
 
     @Test(dataProvider = "CalcNIndelInformativeReadsData")
-    public void testCalcNIndelInformativeReads(final String readBases, final String cigar, final String ref, final int maxIndelSize, final List<Integer> expected ) {
+    public void testCalcNIndelInformativeReads(final String readBases, final String cigar, final byte[] readQuals, final String ref, final int maxIndelSize, final int readStartIntoRef, final List<Integer> expected ) {
         final byte qual = (byte)30;
-        final byte[] quals = Utils.dupBytes(qual, readBases.length());
+        final byte[] quals = readQuals != null ? readQuals : Utils.dupBytes(qual, readBases.length());
+        // on the same read after the first site the result will be cached in the transient attributes, assert the results are the same as those calculated non-transiently.
+        final GATKRead readCache = ArtificialReadUtils.createArtificialRead(readBases.getBytes(), quals, cigar);
 
         for ( int i = 0; i < readBases.getBytes().length; i++ ) {
-            final GATKRead read = ArtificialReadUtils.createArtificialRead(readBases.getBytes(), quals, cigar);
-            final SimpleInterval loc = new SimpleInterval("20", i + 1, i + 1);
-            final ReadPileup pileup = new ReadPileup(loc, Collections.singletonList(read), i);
-            final int actual = model.calcNIndelInformativeReads(pileup, i, ref.getBytes(), maxIndelSize);
-            Assert.assertEquals(actual, (int)expected.get(i), "failed at position " + i);
+            final Pair<Integer, Boolean> readCoordinateForReferenceCoordinate = ReadUtils.getReadCoordinateForReferenceCoordinate(readCache, readCache.getStart() + i, true);
+
+            if (!readCoordinateForReferenceCoordinate.getValue() && readCoordinateForReferenceCoordinate.getKey() != -1) {
+                final GATKRead readNoCache = ArtificialReadUtils.createArtificialRead(readBases.getBytes(), quals, cigar);
+                final SimpleInterval loc = new SimpleInterval("20", i + 1 + readStartIntoRef, i + 1 + readStartIntoRef);
+                final ReadPileup pileupCache = new ReadPileup(loc, Collections.singletonList(readCache), readCoordinateForReferenceCoordinate.getKey());
+                final ReadPileup pileupNoCache = new ReadPileup(loc, Collections.singletonList(readNoCache), ReadUtils.getReadCoordinateForReferenceCoordinate(readNoCache, readNoCache.getStart() + i).getKey());
```

Comment:
Do you have any test cases with ReadPileups containing more than 1 read? If not, is it worth adding some?